Fix gff and scidx parsing

bernt-matthias opened the pull request:
- gff: if there is a `##gff-version` line then respect it
  - it now is analogous to the Gff3 sniffer
  - follow up to https://github.com/galaxyproject/galaxy/pull/12676
- scidx: do not skip the 2nd line

this is fall out of https://github.com/galaxyproject/galaxy/pull/12073 .. found while fixing IUC tools

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

bernt-matthias committed:
fix gff and scidx parsing

gff: if there is a `##gff-version` line then respect it
     it now is analogous to the Gff3 sniffer
scidx: do not skip the 2nd line

gregvonkuster commented:
> For ScIdx:
> 
> * should we add a test file to `lib/galaxy/datatypes/test/` ?
> * I think we could greatly simplify the sniffer to just do this: skip the first line iff it starts with `#`, and then check if the next line is equal to `chrom	index	forward	reverse	value`
> 
> Ping @gregvonkuster

I think that's fine.  FWIW, there is a very limited number of tools that use this datatype, and I'm not sure if they are actively used.

bernt-matthias committed:
fix and simplify scidx sniffer + add test

sniffer was broken: the content of the header line was also checked for being integers which raised an exception:

https://github.com/galaxyproject/galaxy/blob/25fe76a3d68e82ff58f3c35ba86644fe22742430/lib/galaxy/datatypes/interval.py#L1510

now

- 1st line needs to be comment
- 2nd line needs to be the scidx specific headers
- 3rd line needs to be ints (except 1st col) and fwd and rev need to sum
  to value

bernt-matthias committed:
allow empty scidx files

ie only comment + header

bernt-matthias committed:
Reformatting

Co-authored-by: Nicola Soranzo <nicola.soranzo@gmail.com>

bernt-matthias committed:
Really allow cases with only comment and header

Co-authored-by: Nicola Soranzo <nicola.soranzo@gmail.com>

bernt-matthias committed:
split directly with iter_headers

Co-authored-by: Nicola Soranzo <nicola.soranzo@gmail.com>
